core-spi-test/src/main/java/org/ehcache/internal/tier/AuthoritativeTierComputeIfAbsentAndFault.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.internal.tier;
+
+import org.ehcache.exceptions.CacheAccessException;
+import org.ehcache.expiry.Duration;
+import org.ehcache.expiry.Expirations;
+import org.ehcache.function.Function;
+import org.ehcache.internal.TestTimeSource;
+import org.ehcache.spi.cache.tiering.AuthoritativeTier;
+import org.ehcache.spi.test.After;
+import org.ehcache.spi.test.Before;
+import org.ehcache.spi.test.Ignore;
+import org.ehcache.spi.test.SPITest;
+import org.hamcrest.CoreMatchers;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+/**
+ * Test the {@link AuthoritativeTier#computeIfAbsentAndFault(Object, Function)} contract of the
+ * {@link AuthoritativeTier AuthoritativeTier} interface.
+ * <p/>
+ *
+ * @author Aurelien Broszniowski
+ */
+
+public class AuthoritativeTierComputeIfAbsentAndFault<K, V> extends SPIAuthoritativeTierTester<K, V> {
+
+  protected AuthoritativeTier<K, V> tier;
+
+  public AuthoritativeTierComputeIfAbsentAndFault(final AuthoritativeTierFactory<K, V> factory) {
+    super(factory);
+  }
+
+  @Before
+  public void setUp() {
+  }
+
+  @After
+  public void tearDown() {
+    if (tier != null) {
+      tier.close();
+      tier = null;
+    }
+  }
+
+  /**
+   * The goal of this test is to make sure that an entry that was calculated by the computeIfAbsent method
+   * will be evicted with the default behaviour of the tier.
+   */
+  @SPITest
+  public void nonMarkedMappingIsEvictable() {
+    K key = factory.createKey(1);
+
+    tier = factory.newStore(factory.newConfiguration(factory.getKeyType(), factory.getValueType(),
+        1L, null, null, Expirations.noExpiration()));
+
+    try {
+      tier.computeIfAbsent(key, new Function<K, V>() {
+        @Override
+        public V apply(final K k) {
+          return factory.createValue(1L);
+        }
+      });
+
+      for (long seed = 2L; seed < 15000; seed++) {
+        tier.put(factory.createKey(seed), factory.createValue(seed));
+      }
+      assertThat(tier.get(key), is(nullValue()));
+
+    } catch (CacheAccessException e) {
+      System.err.println("Warning, an exception is thrown due to the SPI test");
```

Comment:
That's a sad error handling. Is there a reason why the test method cannot be marked "throws Exception"?